docs/Qualitaetssicherung.tex

```diff
@@ -41,6 +41,25 @@
 \section{Einleitung}
 
 \section{Änderungen in der Implementierung}
+Während der Testphase wurde einige Fehler behoben und Verhaltensweisen verändert um OSIP benutzerfreundlicher zu machen.
+Diese Probleme wurde nicht notwendigerweise durch die Testszenarien aufgedeckt, da die Verhaltensweisen nicht so genau
+spezifiziert sind oder die Szenarien die Fehler nicht aufdecken konnten. Die wichtigsten Änderungen sind die folgenden:
+
+\begin{itemize}
+ \item Vereinfachen von Übersetzungen von OSIP in andere Sprachen, in dem die lokalisierten Fehlermeldungen sämtlicher Exceptions benutzt werden.
+ \item Beheben von Abstürzen, z.b. wenn invalide Werte in den Einstellungsdateien stehen.
+ \item Beheben von Verklemmungen durch zu starke Synchronisierung zwischen Threads.
+ \item Öffnen sämtlicher Dialoge (wie z.B. Alarmmeldungen, Einstellungen oder Überlaufmeldungen) über den anderen Fenstern und model machen dieser Dialoge,
+ so dass OSIP nicht bedienbar ist solange diese nicht geschlossen werden. Das verhindert beispielsweise, dass der Benutzer nach einem Verbindungsverlust
+ die Einstellungen nicht bedienen kann weil der Fehlerdialog im Hintergrund unbemerkt offen ist.
+ \item Verbesserte Logik beim Starten der OPC UA Server und Clients. Wenn ein Server Port mehrfach verwendet wird, wird sofort das
+ Einstellungsfenster geöffnet. Außerdem wurde sichergestellt, dass der Benutzer die Einstellungen in jedem Zustand aufrufen kann.
```

Comment:
Das in jedem Zustand war so eine Art Regression durch die Thread separierung. Die fand aber auch in der aktuellen Phase statt (als bug fix für langsames UI)